underscore.js

```diff
@@ -647,6 +647,20 @@
     return range;
   };
 
+  // Split an **array** into several arrays containing **count** or less elements
+  // of initial array
+  _.chunk = function(array, count) {
+    var result = [];
+
+    if (array == null || count <= 0) return [];
+
+    for (var i = 0, length = array.length; i < length;) {
+      result.push(array.slice(i, i += count));
```

Comment:
We should use `slice.call(array, i, i+= count)` to sidestep overwritten `Array#slice` and open this up to array-likes.